File: leetcode/Minimum_Depth_of_Binary_Tree.java
/**
 * Definition for binary tree
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public int minDepth(TreeNode root) {
        if (root == null) return 0;
        Queue<TreeNode> q = new LinkedList<TreeNode>();
        Queue<Integer> l = new LinkedList<Integer>();
        q.add(root);
        l.add(1);
        while (!q.isEmpty()) {
            TreeNode t = q.poll();
            int cl = l.poll();
            if (t.left == null && t.right == null) {
                return cl;
            }
            if (t.left != null) {
                q.add(t.left);
                l.add(cl + 1);
            }
            if (t.right != null) {
                q.add(t.right);
                l.add(cl + 1);
            }
        }
        return 0;
    }
}

File: leetcode/Minimum_Depth_of_Binary_Tree.cpp
/**
 * Definition for binary tree
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int minDepth(TreeNode *root) {
        if (root == NULL)
            return 0;
        queue<TreeNode*> node_queue;
        node_queue.push(root);
        queue<int> depth_queue;
        depth_queue.push(1);
        while (!node_queue.empty()) {
            TreeNode *node = node_queue.front();
            node_queue.pop();
            int depth = depth_queue.front();
            depth_queue.pop();
            if (node->left == NULL && node->right == NULL)
                return depth;
            if (node->left != NULL) {
                node_queue.push(node->left);
                depth_queue.push(depth + 1);
            }
            if (node->right != NULL) {
                node_queue.push(node->right);
                depth_queue.push(depth + 1);
            }
        }
    }
};
